boa/util/disk_cache.py

```diff
@@ -66,8 +66,9 @@ def caching_lookup(self, string, func):
                 return pickle.loads(f.read())
         except OSError:
             res = func()
-            tid = threading.get_ident()
-            tmp_p = p.with_suffix(f".{tid}.unfinished")
+            # note: how portable is os.getpid()? should we use uuid generator?
```

Comment:
getpid seems portable enough
```suggestion
            # use process ID and thread ID to avoid race conditions
```